[Doc] Improve a bit for developer build guidance (#1030)
taichi-dev/taichi: docs/dev_install.rst

    mkdir build
    cd build
    cmake ..

>     make -j$(nproc)

Reviewer: note `nproc` is not avaialble on macOS. Might want to try something like `$(nproc 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || echo 8)` to make it portable and have a default fallback?

Author: Great suggestion to apply!
I do not have a mac machine but ```2>/dev/nul``` ensures no error will be inputted, and ```|| echo 8``` does ensure the fallback. 
The safest way is ``` echo 1```,nonetheless, if 8 has been fine so far I assume the developers who indeed visit this page will find it easy to determine a proper job number.

>     make -j $(nproc 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || echo 8)

Reviewer 2: Thanks for updating this! I actually think `make -j 8` is better, given the newer version is too scary. 
Note that having threads more than #CPU cores isn't really a disaster, except for out-of-memory of cache contention - which are unlikely to happen. Even if those happens, I believe most users know that they should simply reduce the number `8`.

Author: Thanks for your patience, 

lol its too lengthy indeed, but here is what our company does which I find a bit helpful: write a ./Allmake script to handle all these lengthy input flags/variables at higher level, and also treats differently depending on the operating system. So what people have to do is simply `` chmod +x Allmake `` then `` ./Allmake``

An example:
```
if [ "$(uname)" == "Darwin" ]; then
    echo "Mac is currectly not supported"       
elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
    echo "Detected plattform: Linux not Mac"
    echo "Compiling Debug mode"
    echo "Compiling Release mode"
elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ] || [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
    echo "Detected plattform: Win"
    echo "Compiling Debug mode"
    echo "Compiling Release mode"
fi
```

We can create later if needed, but for now I will just reset to `` -j 8 ``

    mkdir build
    cd build
    cmake ..

>     # if you donot set clang as the default compiler in Linux

Reviewer 3: nit:
```suggestion
    # if you do not set clang as the default compiler
```
1. Please use `do not` for clarity.
2. Adding `clang` to `PATH` can be also applied to OS X and Windows.

Author: Thanks! please give me a while to learn your 2) suggestion then I will commit these suggestions.

    mkdir build
    cd build
    cmake ..
    # if you do not set clang as the default compiler
    # use the line below:
    #   cmake .. -DCMAKE_CXX_COMPILER=clang-8
    #
    # Alternatively, if you would like to set clang as the default compiler

>     # CMAKE honors variables ``CC`` and ``CXX`` upon detecting the C and C++ compiler to use

Reviewer 2: ```suggestion
    # On Unix CMake honors environment variables $CC and $CXX upon deciding which C and C++ compilers to use
```
